src/rust/logger/src/receiver.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use crate::*;
+
+/// A `LogReceiver` is the receiving end of the log queues. It should be owned
+/// by the thread which will be flushing the log messages to their respective
+/// outputs. It is expected that the `flush()` function will be periodically
+/// called to make room for new log messages.
+pub struct LogReceiver {
+    pub(crate) debug: FileLogReceiver,
```

Comment:
I think it's OK to make debug logging optional too. We normally don't do that but it can be used to see if logging introduces any meaningful overhead. When I first put logging into Pelikan-C I saw it popping up in profiling, until I changed the macro implementation to move level comparison logic outside the function body.